src/fiber/execution_context/multi_threaded.cr

```diff
@@ -244,7 +241,7 @@ module Fiber::ExecutionContext
         index = @threads.size
         return if index == @size.end # check again
```

Comment:
This doesn't account for range exclusivity - I'd suggest to normalize the range in the initializer to avoid the need for checking this every time.